tools/prow-config-updater/updater.go

```diff
@@ -0,0 +1,222 @@
+		return fmt.Errorf("error updating staging Prow configs: %v", err)
+	}
+	// Create pull request in the fork repository for the testing of staging Prow.
+	fpr, err := cli.githubmainhandler.createForkPullRequest(cli.forkOrgName)
+	if err != nil {
+		return fmt.Errorf("error creating pull request in the fork repository: %v", err)
+	}
+	// Create comment on the fork pull request to get it tested by staging Prow and merged.
+	forkprnumber := *fpr.Number
+	if err = cli.githubcommenter.tryMergeForkPullRequest(cli.forkOrgName, forkprnumber); err != nil {
+		return fmt.Errorf("error creating comment on the fork pull request: %v", err)
+	}
+	// Wait for the fork pull request to be automatically merged by staging Prow.
+	if err := cli.githubmainhandler.waitForForkPullRequestMerged(cli.forkOrgName, forkprnumber); err != nil {
+		return fmt.Errorf("error waiting for the fork pull request to be merged: %v", err)
+	}
+	return nil
+}
+
+// Filter out all files that are under the given paths.
+func collectRelevantFiles(files []string, paths []string) []string {
+	rfs := make([]string, 0)
+	for _, f := range files {
+		for _, p := range paths {
+			if strings.HasPrefix(f, p) {
+				rfs = append(rfs, f)
+			}
+		}
+	}
+	return rfs
+}
+
+// Run the `make` command to update Prow configs.
+// path is the Prow config root folder.
+func (cli *Client) doProwUpdate(env config.ProwEnv) ([]string, error) {
+	relevantFiles := make([]string, 0)
+	updateCommand := ""
+	switch env {
+	case config.ProdProwEnv:
+		relevantFiles = append(relevantFiles, collectRelevantFiles(cli.files, config.ProdProwConfigPaths)...)
+		updateCommand = config.UpdateProdProwCommand
+	case config.StagingProwEnv:
+		relevantFiles = append(relevantFiles, collectRelevantFiles(cli.files, config.StagingProwConfigPaths)...)
+		updateCommand = config.UpdateStagingProwCommand
+	default:
+		return nil, fmt.Errorf("unsupported Prow environement: %q, cannot make the update", env)
+	}
+	if len(relevantFiles) != 0 {
+		if err := helpers.Run(
+			fmt.Sprintf("Updating Prow configs with command %q", updateCommand),
+			func() error {
+				out, err := cmd.RunCommand(updateCommand)
+				log.Println(out)
+				return err
+			},
+			cli.dryrun,
+		); err != nil {
+			return nil, fmt.Errorf("error updating Prow configs for %q environment: %v", env, err)
+		}
+	}
+
+	// For production Prow, we also need to update Testgrid config if it's changed.
+	// TODO(chizhg): this will be removed after we get rid of Testgrid config file by moving to ProwJob annotation.
+	if env == config.ProdProwEnv {
+		tfs := collectRelevantFiles(cli.files, []string{config.ProdTestgridConfigPath})
+		if len(tfs) != 0 {
+			relevantFiles = append(relevantFiles, tfs...)
+			if err := helpers.Run(
+				fmt.Sprintf("Updating Testgrid config with command %q", config.UpdateTestgridCommand),
+				func() error {
+					out, err := cmd.RunCommand(config.UpdateTestgridCommand)
+					log.Println(out)
```

Comment:
debug?

Reply:
The `log` library does not have `debug` method :( If we want better logging, we may switch to `logrus` as what kubernetes test-infra does.